### PR TITLE
Implement reset method in BlockDataCollector

### PR DESCRIPTION
**Pull request by mzstic**
Since Symfony 3.4 classes implementing DataCollectorInterface should implement reset() method.

I am targeting 3.x branch, because I'm only adding new method. No BC changes. There's no reason for waiting for new major release.

Closes https://github.com/sonata-project/SonataAdminBundle/issues/4699
Didn't know how else should I refer to issue in another bundle...

## Changelog

```markdown
### Added
- Implement reset method in `BlockDataCollector` to be compatible with Symfony 3.4
```
## Subject
Added one method to BlockDataCollector to implement new version of DataCollectorInterface correctly.

### CODE REVIEW

**Comment by greg0ire** [comment]
Please move the subject out of the changelog. Also, the collector is not an interface, you cannot implement it. Please change the changelog message accordingly

**Comment by mzstic** [comment]
@greg0ire I added unit test for BlockDataCollector. I'm just not sure ho style CI works... I understand there are some code style errors in files I changed, but style CI reports some errors in files I haven't touched. I'm just not sure if I should fix all of it or just the files I actually changed...?

**Comment by greg0ire** [comment]
Either wait for us to fix those new issues or do it in a separate commit. These come from a new fixer in php-cs-fixer.

File: Tests/Profiler/DataCollector/BlockDataCollectorTest.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+        $expectedRealBlocks = [
+            'test2' => [
+                'type' => 'another_type',
+            ]
+        ];
+        $this->assertEquals($expectedRealBlocks, $blockDataCollector->getRealBlocks());
+
+        $blockDataCollector->reset();
+        $this->assertEquals([], $blockDataCollector->getEvents());
```
Comment (OskarStark) [inline Tests/Profiler/DataCollector/BlockDataCollectorTest.php]: assertEmpty would be better IMO

File: Tests/Profiler/DataCollector/BlockDataCollectorTest.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\BlockBundle\Tests\Profiler\DataCollector;
+
+use PHPUnit\Framework\TestCase;
+use Sonata\BlockBundle\Profiler\DataCollector\BlockDataCollector;
+
+class BlockDataCollectorTest extends TestCase
```
Comment (Soullivaneuh) [inline Tests/Profiler/DataCollector/BlockDataCollectorTest.php]: * `final`
* Please add your author tag.

**Comment by greg0ire** [comment]
Ping @mzstic !

File: Tests/Profiler/DataCollector/BlockDataCollectorTest.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\BlockBundle\Tests\Profiler\DataCollector;
+
+use PHPUnit\Framework\TestCase;
+use Sonata\BlockBundle\Profiler\DataCollector\BlockDataCollector;
+
+final class BlockDataCollectorTest extends TestCase
```
Comment (OskarStark) [inline Tests/Profiler/DataCollector/BlockDataCollectorTest.php]: please add `@author` tags to the newly created classes

**Comment by OskarStark** [comment]
I updated your changelog @mzstic and added 2 commits to your PR, so please pull this feature branch before force push.

Please rebase this, this should make StyleCI happy 😄

**Comment by jordisala1991** [comment]
fixed in https://github.com/sonata-project/SonataBlockBundle/pull/439

**Pull request closed without merging**